CIP-transaction-inputs-as-unordered-set/README.md

```diff
@@ -0,0 +1,140 @@
+    find (\TxInInfo{txInInfoOutRef} -> txInInfoOutRef == txOutRef) txInfoInputs
+findOwnInput _ = Nothing
+ ```
+
+ ```haskell
+getContinuingOutputs :: ScriptContext -> [TxOut]
+getContinuingOutputs ctx | Just TxInInfo{txInInfoResolved=TxOut{txOutAddress}} <- findOwnInput ctx = filter (f txOutAddress) (txInfoOutputs $ scriptContextTxInfo ctx)
+    where
+        f addr TxOut{txOutAddress=otherAddress} = addr == otherAddress
+getContinuingOutputs _ = traceError "Lf" -- "Can't get any continuing outputs"
+```
+
+```haskell
+validatorA :: BuiltinData -> BuiltinData -> ScriptContext -> Bool
+validatorA _datum _redeemer context =
+  validateWithInputOutput input output
+  where
+    Just (TxInInfo {txInInfoResolved = input}) = findOwnInput context
+    [output] = getContinuingOutputs context
+    validateWithInputOutput _ _ = True
+```
+
+Furthermore, maintaining the order of inputs from the submitted transaction will facilitate the utilization of index redeemer patterns that many projects are transitioning to, as illustrated below.
+
+```haskell
+validatorIndex :: BuiltinData -> Integer -> ScriptContext -> Bool
+validatorIndex _datum index ScriptContext {scriptContextTxInfo = txInfo, scriptContextPurpose = Spending txOutRef} =
+  validateInputOutput input output && txOutRef == intxOutRef
+  where
+    inputs = txInfoInputs txInfo
+    outputs = txInfoOutputs txInfo
+    TxInInfo {txInInfoOutRef = intxOutRef, txInInfoResolved = input} = inputs P.!! index
+    output = outputs P.!! index
+    validateWithInputOutput _ _ = True
+```
+
+Given the deterministic nature of the extended UTXO model, the determination of inputs and outputs can already be achieved through the off-chain code, eliminating the necessity to find or traverse inputs and outputs within the validator.
+
+## Specification
+As per protocol specifications, the transaction body is structured as follows:
+
+```
+transaction_body =
+  { 0 : set<transaction_input>    ; inputs
+  , 1 : [* transaction_output]
+  , 2 : coin                      ; fee
+  , ? 3 : uint                    ; time to live
+  , ? 4 : [* certificate]
+  , ? 5 : withdrawals
+  , ? 6 : update
+  , ? 7 : auxiliary_data_hash
+  , ? 8 : uint                    ; validity interval start
+  , ? 9 : mint
+  , ? 11 : script_data_hash
+  , ? 13 : set<transaction_input> ; collateral inputs
+  , ? 14 : required_signers
+  , ? 15 : network_id
+  , ? 16 : transaction_output     ; collateral return; New
+  , ? 17 : coin                   ; total collateral; New
+  , ? 18 : set<transaction_input> ; reference inputs; New
+  }
+```
+
+Specifically, the inputs and reference inputs are currently represented as a set:
+```
+0 : set<transaction_input>    ; inputs
+```
+```
+18 : set<transaction_input> ; reference inputs; New
+```
+
+The proposed solution suggests modifying the inputs to an unordered set format:
+```
+0 : unordered_set<transaction_input>    ; inputs
```

Comment:
See my earlier comment
```suggestion
0 : oset<transaction_input>    ; inputs
```

Reply:
I see this makes sense, I didn't understand the meaning of `oset`which in haskell terms means , `An OSet behaves much like a Set but remembers the order in which the elements were originally inserted.` taken from this hackage link [Oset](https://hackage.haskell.org/package/oset-0.4.0.1/docs/Data-Set-Ordered-OSet.html)

Reply:
Yep, except we use our own implementation of `OSet`, since we need to ensure strictness of elements